src/cluster_legacy.c

```diff
@@ -5961,11 +5956,23 @@ int clusterParseSetSlotCommand(client *c, int *slot_out, clusterNode **node_out,
                 return 0;
             }
         }
+        if (c->argc > 5) optarg_pos = 5;
     } else {
         addReplyError(c, "Invalid CLUSTER SETSLOT action or number of arguments. Try CLUSTER HELP");
         return 0;
     }
 
+    /* Process optional arguments */
+    for (int i = optarg_pos; i < c->argc; i++) {
+        if (!strcasecmp(c->argv[i]->ptr, "timeout")) {
+            if (i + 1 >= c->argc) {
+                addReplyError(c, "Missing timeout value");
```

Comment:
Most checks like this simply use this shared error message:

```suggestion
                addReplyErrorObject(c, shared.syntaxerr);
```

Reply:
The syntax error string is too generic. Also `getTimeoutFromObjectOrReply()` returns specific errors too. I think it makes sense to go along with that.

Reply:
OK, I accept you preference. :-)

My thinking was that there's enough syntax information available in our command JSON files to let clients validate the syntax on the client side. Our errors don't need to be verbose docs.

Reply:
I would agree with you if it was not for getTimeoutFromObjectOrReply :)